src/ui/components/shared/Sharing.js

```diff
@@ -0,0 +1,336 @@
+import React, { useEffect, useState } from "react";
+import { connect } from "react-redux";
+import classnames from "classnames";
+import { gql, useLazyQuery, useQuery, useMutation } from "@apollo/client";
+import { selectors } from "ui/reducers";
+import { actions } from "ui/actions";
+import Modal from "./Modal";
+import Loader from "./Loader";
+import { features } from "ui/utils/prefs";
+
+import "./Sharing.css";
+import { bindActionCreators } from "redux";
+
+// This is a contrived demo since I don't have a new recording to experiment with. Once
+// the Hasura table and dispatcher are synced again, I should:
+
+// 1) GET_O_AND_C should take just one argument $recording_id: uuid
+// 2) We should only match the recording_id, and not the id in the GOAC collaborators args
+// 3) Remove the fakeRecordingId that we pass down from the SharingModal component
+
+// TODOS
+// Add a way to merge and update the cache upon deletion
+// Trigger a refetch of collaborators upon mutation
+// Take out the sharingmodal logic and styles from modal and reuse it somehow with userprompt
+// Figure out what the unified userprompt/modal would look like
+
+const GET_OWNER_AND_COLLABORATORS = gql`
+  query GetOwnerAndCollaborators($idUuid: uuid, $idString: String) {
+    collaborators(
+      where: {
+        recording: { recording_id: { _eq: $idString }, _or: { recording_id: { _eq: $idString } } }
+      }
+    ) {
+      user {
+        auth_id
+        email
+        id
+        name
+        nickname
+        picture
+      }
+      user_id
+      recording_id
+    }
+    recordings(where: { recording_id: { _eq: $idString } }) {
+      user {
+        auth_id
+        email
+        id
+        name
+        nickname
+        picture
+      }
+      id
+      is_private
+    }
+  }
+`;
+
+// Older recordings have a different recordingId vs id. For, Newer recordings it's the same.
```

Comment:
lets just ignore older recordings going forward. We can always update these IDs later